### PR TITLE
WinHTTP: Try to use TLS1.3

### PR DESCRIPTION
**Pull request by csware**
Tested on Windows 7 and 10 (of course WIndows 7 does not support TLS1.3; for WIndows 10 TLS1.3 ATM needs to be enabled in Internet Options and manually as of 2004 cf. https://stackoverflow.com/a/59210166/3906760)

### CODE REVIEW

File: src/transports/winhttp.c [hunk 1]
```diff
@@ -788,14 +793,21 @@ static int winhttp_connect(
...
+	 * fail; if TLS 1.2 or 1.3 support is not available for some reason,
 	 * ignore the failure (it will keep the default protocols).
 	 */
-	WinHttpSetOption(t->session,
+	if (WinHttpSetOption(t->session,
+	    WINHTTP_OPTION_SECURE_PROTOCOLS,
+	    &protocols,
+	    sizeof(protocols)) == FALSE) {
+	    protocols &= ~WINHTTP_FLAG_SECURE_PROTOCOL_TLS1_3;
+	    MessageBox(NULL, "djdjd", "kjfjf", 0);
```
Comment (ethomson) [inline src/transports/winhttp.c:805]: Looks like you've left some testing/debugging code in.

Comment (csware) [reply src/transports/winhttp.c:805]: Oops, thanks!

Comment (csware) [reply src/transports/winhttp.c]: I already wondered why the test timed out...

Thanks, fixed.

**Review by ethomson — APPROVED** [review]
Great, thanks!

**Pull request merged**